src/urllib3/poolmanager.py

```diff
@@ -116,6 +123,10 @@ def _default_key_normalizer(key_class, request_context):
         if field not in context:
             context[field] = None
 
+    # Default key blocksize to 16384 if missing from the context
+    if _CONN_CLS_ACCEPTS_BLOCKSIZE and context.get("key_blocksize") is None:
```

Comment:
If `context` is a dict, maybe using `dict.setdefault` would be better? 

```
if _CONN_CLS_ACCEPTS_BLOCKSIZE:
    context.setdefault("key_blocksize", 16384)
```

would be the same as 
```
if _CONN_CLS_ACCEPTS_BLOCKSIZE and "key_blocksize" not in context:
    context["key_blocksize"] = 16384
```

This won't overwrite an explicitly set `None` of course, so if that's desired then current behavior can stay.